frontend/src/scenes/experiments/ExperimentView/SecondaryMetricsTable.tsx

```diff
@@ -135,8 +90,23 @@ export function SecondaryMetricsTable({
                             </LemonButton>
                         </div>
                     </>
-                }
-            >
+                )
+            }
+        >
+            {isModalReadOnly ? (
+                <div>
+                    <ExperimentInsightCreator insightProps={insightProps} />
```

Comment:
Here we need -> https://github.com/PostHog/posthog/blob/secondary-metrics-table/frontend/src/scenes/experiments/ExperimentResult.tsx#L215 the valid metric. Potentially worth refactoring `ExperimentResults` component that you're using for the primary metric to make it work here